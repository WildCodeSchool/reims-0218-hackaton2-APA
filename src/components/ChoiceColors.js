import React, { Component } from 'react'
import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap'

export class ChoiceEye extends Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      dropdownOpen: false
    };
  }

  toggle() {
    this.setState(prevState => ({
      dropdownOpen: !prevState.dropdownOpen
    }));
  }

  render() {
    return (
      <Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
        <DropdownToggle caret>
          Eye Color
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem>Blue</DropdownItem>
          <DropdownItem>Red</DropdownItem>
          <DropdownItem>Green</DropdownItem>
          <DropdownItem>Black</DropdownItem>
          <DropdownItem>Yellow</DropdownItem>
        </DropdownMenu>
      </Dropdown>
    );
  }
}

export class ChoiceSkin extends Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      dropdownOpen: false
    };
  }

  toggle() {
    this.setState(prevState => ({
      dropdownOpen: !prevState.dropdownOpen
    }));
  }

  render() {
    return (
      <Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
        <DropdownToggle caret>
          Skin Color
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem>Blue</DropdownItem>
          <DropdownItem>Red</DropdownItem>
          <DropdownItem>Green</DropdownItem>
          <DropdownItem>Black</DropdownItem>
          <DropdownItem>Yellow</DropdownItem>
        </DropdownMenu>
      </Dropdown>
    );
  }
}

