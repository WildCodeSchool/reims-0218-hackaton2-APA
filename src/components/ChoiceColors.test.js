import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {ChoiceEye, ChoiceSkin} from "./ChoiceColors";

it('renders Eye without crashing', () => {
  const divEye = document.createElement('div');
  ReactDOM.render(<ChoiceEye />, divEye);
  ReactDOM.unmountComponentAtNode(divEye);
});

it('renders Skin without crashing', () => {
  const divSkin = document.createElement('div');
  ReactDOM.render(<ChoiceSkin />, divSkin);
  ReactDOM.unmountComponentAtNode(divSkin);
});

it('renders Eye correctly', () => {
  
  const props = {

     colors : ["Blue", "Red", "Green","Black","Yellow"]
  }
  
  const treeEye = renderer
    .create(<ChoiceEye {...props} />)
    .toJSON();
  expect(treeEye).toMatchSnapshot();
});

it('renders Skin correctly', () => {
  
  const props = {

     colors : ["Blue", "Red", "Green","Black","Yellow"]
  }
  
  const treeSkin = renderer
    .create(<ChoiceSkin {...props} />)
    .toJSON();
  expect(treeSkin).toMatchSnapshot();
});

