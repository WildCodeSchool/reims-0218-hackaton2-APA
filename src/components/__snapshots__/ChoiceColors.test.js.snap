// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Eye correctly 1`] = `
<div
  className="dropdown"
  onKeyDown={[Function]}
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-secondary"
    onClick={[Function]}
    type="button"
  >
    Eye Color
  </button>
  <div
    aria-hidden={true}
    className="dropdown-menu"
    role="menu"
    tabIndex="-1"
    x-placement={undefined}
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Blue
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Red
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Green
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Black
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Yellow
    </button>
  </div>
</div>
`;

exports[`renders Skin correctly 1`] = `
<div
  className="dropdown"
  onKeyDown={[Function]}
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-secondary"
    onClick={[Function]}
    type="button"
  >
    Skin Color
  </button>
  <div
    aria-hidden={true}
    className="dropdown-menu"
    role="menu"
    tabIndex="-1"
    x-placement={undefined}
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Blue
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Red
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Green
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Black
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      Yellow
    </button>
  </div>
</div>
`;
